pandas/core/arrays/numpy_.py

```diff
@@ -263,12 +263,8 @@ def __setitem__(self, key, value) -> None:
         value = extract_array(value, extract_numpy=True)
 
         key = check_array_indexer(self, key)
-        scalar_key = lib.is_scalar(key)
         scalar_value = lib.is_scalar(value)
 
-        if not scalar_key and scalar_value:
-            key = np.asarray(key)
```

Comment:
@TomAugspurger  do you remember why this was needed? 
(I can also do a `and not isinstance(key, slice)` to be more conservative, but can't directly come up with a reason to keep it)

Reply:
I don’t recall.